pkg/component/gardener/apiserver/deployment.go

```diff
@@ -165,3 +168,41 @@ func (g *gardenerAPIServer) deployment(
 
 	return deployment
 }
+
+func injectWorkloadIdentitySettings(deployment *appsv1.Deployment, secret *corev1.Secret) {
+	var (
+		mountPath = "/etc/gardener-apiserver/workload-identity/signing"
+		fileName  = "key.pem"
+	)
+
+	deployment.Spec.Template.Spec.Containers[0].Args = append(
+		deployment.Spec.Template.Spec.Containers[0].Args,
+		fmt.Sprintf("--workload-identity-signing-key-file=%s/%s", mountPath, fileName),
+	)
+
+	deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts,
+		corev1.VolumeMount{
+			Name:      "gardener-apiserver-workload-identity",
```

Comment:
Nit: define a constant for this volume name as well and use it here and in L194

Reply:
Done